alibi_detect/cd/pytorch/classifier.py

```diff
@@ -134,30 +149,35 @@ def score(self, x: np.ndarray) -> Tuple[float, float]:
 
         Returns
         -------
-        p-value, and a notion of distance between the trained classifier's out-of-fold performance
-        and that which we'd expect under the null assumption of no drift.
+        p-value, a notion of distance between the trained classifier's out-of-fold performance
+        and that which we'd expect under the null assumption of no drift,
+        the classification labels (0=reference data, 1=test data) and out-of-fold classifier predictions.
         """
         x_ref, x = self.preprocess(x)
-        n_ref, n_cur = x_ref.shape[0], x.shape[0]
-
+        n_ref, n_cur = len(x_ref), len(x)
         x, y, splits = self.get_splits(x_ref, x)
 
         # iterate over folds: train a new model for each fold and make out-of-fold (oof) predictions
         preds_oof_list, idx_oof_list = [], []
         for idx_tr, idx_te in splits:
-            x_tr, y_tr, x_te = x[idx_tr], y[idx_tr], x[idx_te]
-            ds_tr = TensorDataset(torch.from_numpy(x_tr), torch.from_numpy(y_tr))
-            dl_tr = DataLoader(ds_tr, **self.dl_kwargs)  # type: ignore
+            y_tr = y[idx_tr]
+            if isinstance(x, np.ndarray):
+                x_tr, x_te = x[idx_tr], x[idx_te]
+            elif isinstance(x, list):
+                x_tr, x_te = [x[_] for _ in idx_tr], [x[_] for _ in idx_te]
+            else:
+                raise TypeError(f'x needs to be of type np.ndarray or list and not {type(x)}.')
+            ds_tr = self.dataset(x_tr, y_tr)
+            dl_tr = self.dataloader(ds_tr)
             model = deepcopy(self.model)
             train_args = [model, self.loss_fn, dl_tr, self.device]
             trainer(*train_args, **self.train_kwargs)  # type: ignore
-            preds = predict_batch(x_te, model.eval(), device=self.device, batch_size=self.dl_kwargs['batch_size'])
+            preds = self.predict_fn(x_te, model.eval())
             preds_oof_list.append(preds)
             idx_oof_list.append(idx_te)
         preds_oof = np.concatenate(preds_oof_list, axis=0)
         probs_oof = softmax(preds_oof, axis=-1) if self.preds_type == 'logits' else preds_oof
         idx_oof = np.concatenate(idx_oof_list, axis=0)
         y_oof = y[idx_oof]
-
         p_val, dist = self.test_probs(y_oof, probs_oof, n_ref, n_cur)
-        return p_val, dist
+        return p_val, dist, y_oof, probs_oof[:, 1]
```

Comment:
Simply returning `y_oof` and `probs_oof` doesn't seem particularly useful to me. It is a mix of reference and test instances ordered in a way that doesn't correspond to either the reference or test window? So even if high scores are observed there's no way to connect the score to the corresponding instances?

Reply:
`y_oof` contains the labels (0=reference, 1=test) and `probs_oof` contains the scores in the same order, so it does correspond to reference and test window I think and could be used downstream to e.g. plot any way you see fit?

Reply:
Ah ok, I see, you meant to tie back to the order of the original reference and test batch, good point.